packages/compiler-vapor/__tests__/compile.spec.ts

```diff
@@ -193,7 +193,49 @@ describe('compile', () => {
       expect(code).contains('_unref(foo)[key.value+1]()')
     })
 
-    // TODO: add more test for expression parsing (v-on, v-slot, v-for)
+    test('v-on', () => {
+      const code = compile(`<div v-on:click="e => fn(e,'str',...args)" />`, {
+        inline: true,
+      })
+      expect(code).matchSnapshot()
+      expect(code).contains(
+        '_delegate(n0, "click", () => e => fn(e,\'str\',...args))',
+      )
+    })
+
+    test('v-for', () => {
+      const code = compile(
+        `<div v-for="list in lists"><div v-for="child in list.children"> {{ child }} </div></div>`,
+        {
+          inline: true,
+        },
+      )
+      expect(code).matchSnapshot()
+    })
+
+    test('v-slot', () => {
+      const code = compile(
+        `
+    <script setup lang="ts">
+    import { Foo, Bar, Baz, Qux, Fred } from './x'
+    const a = 1
+    function b() {}
+    </script>
```

Comment:
The compile function in compiler-vapor only processes the code in the `<template>`, and `<script setup lang="ts">` in the test case may be beyond its scope.